processor/src/main/kotlin/permissions/dispatcher/processor/impl/kotlin/KotlinBaseProcessorUnit.kt

```diff
@@ -105,10 +113,13 @@ abstract class KotlinBaseProcessorUnit(val messager: Messager) : KtProcessorUnit
 
         // Delegate method body generation to implementing classes
         addWithPermissionCheckBody(builder, method, rpe)
+        if (isDeprecated()) {
```

Comment:
we can't annotate file itself as Deprecated in Kotlin so I added it to every `checkWithPermission` method